spectral_cube/stokes_spectral_cube.py

```diff
@@ -193,70 +225,97 @@ def _new_cube_with(self, stokes_data=None,
+
+        # Helper to create SpectralCube from array
+        def make_cube(array, template_key):
+            return SpectralCube(array, wcs=self._wcs, mask=self._stokes_data[template_key].mask)
+
+        # Transformations
+        if self.stokes_type == "FEED_LINEAR" and stokes_basis == "Sky":
+            XX = get_data('XX')
+            YY = get_data('YY')
+            XY = get_data('XY')
+            YX = get_data('YX')
+            I = (XX + YY) / 2
+            Q = (XX - YY) / 2
+            U = (XY + YX) / 2
+            V = 1j * (XY - YX) / 2
+            data = {
+                'I': make_cube(I, 'XX'),
+                'Q': make_cube(Q, 'XX'),
+                'U': make_cube(U, 'XX'),
+                'V': make_cube(V, 'XX'),
+            }
+        elif self.stokes_type == "FEED_CIRCULAR" and stokes_basis == "Sky":
+            RR = get_data('RR')
+            LL = get_data('LL')
+            RL = get_data('RL')
+            LR = get_data('LR')
+            I = (RR + LL) / 2
+            Q = (RL + LR) / 2
+            U = 1j * (RL - LR) / 2
+            V = (RR - LL) / 2
+            data = {
+                'I': make_cube(I, 'RR'),
+                'Q': make_cube(Q, 'RR'),
+                'U': make_cube(U, 'RR'),
+                'V': make_cube(V, 'RR'),
+            }
+        elif self.stokes_type == "SKY_STOKES" and stokes_basis == "Linear":
+            # Sky (I,Q,U,V) to Linear (XX, XY, YX, YY) with 1/2 factor and correct sign
+            I = get_data('I')
+            Q = get_data('Q')
+            U = get_data('U')
+            V = get_data('V')
+            XX = (I + Q) / 2
+            YY = (I - Q) / 2
+            XY = (U + 1j * V) / 2  # sign flip
+            YX = (U - 1j * V) / 2  # sign flip
+            data = {
+                'XX': make_cube(XX, 'I'),
+                'XY': make_cube(XY, 'I'),
+                'YX': make_cube(YX, 'I'),
+                'YY': make_cube(YY, 'I'),
+            }
+        elif self.stokes_type == "SKY_STOKES" and stokes_basis == "Circular":
+            # Sky (I,Q,U,V) to Circular (RR, RL, LR, LL) with 1/2 factor and correct sign
+            I = get_data('I')
+            Q = get_data('Q')
+            U = get_data('U')
+            V = get_data('V')
+            RR = (I + V) / 2
+            LL = (I - V) / 2
+            RL = (Q + 1j * U) / 2  # sign flip
+            LR = (Q - 1j * U) / 2  # sign flip
+            data = {
+                'RR': make_cube(RR, 'I'),
+                'RL': make_cube(RL, 'I'),
+                'LR': make_cube(LR, 'I'),
+                'LL': make_cube(LL, 'I'),
+            }
+        elif stokes_basis == "Sky" and self.stokes_type == "SKY_STOKES":
+            data = self._stokes_data
+        elif stokes_basis == "Linear" and self.stokes_type == "FEED_LINEAR":
+            data = self._stokes_data
+        elif stokes_basis == "Circular" and self.stokes_type == "FEED_CIRCULAR":
+            data = self._stokes_data
```

Comment:
a bit redundant...

Reply:
Well it is to return the same object when the requested transform is the same as a the basis of the cube. Should just warn instead ?

Reply:
I mean, this could all be one `elif` block.... it's fine as is, I guess, since it's readable.